File: r/leap/leap.R
# Name: leap.R
# Purpose: exercism, R track, Leap
# Written by Z Knight, 2019.10.05
#  Modified to and and or instead of nested loops; ZK, 2019.10.05
#
leap <- function(year) {
  if (year %% 4 == 0 & (year %% 100 != 0 | year %% 400 == 0)) {
    return(TRUE)
  }
  return(FALSE)
}

File: javascript/leap/leap.js
/*
 * Name: leap.js
 * Purpose: exercism, javascript track, leap
 * by Z Knight, 2019.10.11
 *   Simplified to single line; ZK, 2020.03.29
 */

export const isLeap = (year) => {
  return (year %4 == 0 && ( year %100 != 0 || year %400 == 0 ));
};

